src/components/register/Register.vue

```diff
@@ -0,0 +1,66 @@
+<script lang="ts">
+import { Ref, ref } from "vue";
+import { registerUser } from "@/services/User.service";
+import router from "@/router";
+import { setAuthToken } from "@/services/Auth.service";
+
+// The state of the register page
+export interface RegisterPageState {
+  first_name: Ref<string>;
+  last_name: Ref<string>;
+  username: Ref<string>;
+  password: Ref<string>;
+  confirm_password: Ref<string>;
+  submit: () => Promise<void>;
+}
+
+export function useRegisterPage(): RegisterPageState {
+  const first_name = ref<string>("");
+  const last_name = ref<string>("");
+  const username = ref<string>("");
+  const password = ref<string>("");
+  const confirm_password = ref<string>("");
+
+  // Parent component will handle the submission of the form
+  async function submit() {
+    if (password.value !== confirm_password.value) {
+      console.error("Passwords do not match");
+      return;
+    }
+
+    const UserRegistrationInfo = {
+      first_name: first_name.value,
+      last_name: last_name.value,
+      username: username.value,
+      password: password.value,
+    };
+    const response = await registerUser(UserRegistrationInfo);
+    setAuthToken(response["access_token"]);
```

Comment:
No need for this line, register will just create user and redirect to login (where they'll get the auth token)